src/content/blog/hybrid-rendering.mdx

```diff
@@ -0,0 +1,114 @@
+---
+title: 'Unlock New Possibilities with Hybrid Rendering'
+description: 'New in Astro 2.0: Hybrid Rendering unlocks the best of both worlds—choose between the performance of static pages or the full flexibility of server rendering for each page of your website.'
+publishDate: 'January 26, 2023'
+authors:
+    - nate
+socialImage: '/assets/blog/hybrid-rendering/social.jpg?v=1'
+coverImage: '/assets/blog/hybrid-rendering/cover.jpg?v=1'
+lang: 'en'
+---
+
+import BlogContentImage from '/src/components/BlogContentImage.astro'
+// Images
+import staticBuild from '/src/assets/blog/hybrid-rendering/static.png'
+import serverBuild from '/src/assets/blog/hybrid-rendering/server.png'
+import hybridBuild from '/src/assets/blog/hybrid-rendering/hybrid.png'
+
+Since the initial launch of Astro 1.0, developers have been able to choose between generating [static (SSG)](https://docs.astro.build/en/core-concepts/routing/#static-ssg-mode) or [server (SSR)](https://docs.astro.build/en/guides/server-side-rendering/) output. Static websites offer incredible performance by building your pages ahead-of-time. Servers leverage their dynamic power to generate HTML on-demand, unique for every request.
+
+Until now, each Astro project has required an all-or-nothing decision—is this fully static or should I deploy a server?
+
+[Astro 2.0](/astro-2) unlocks the best of both worlds with **Hybrid Rendering**. Choose between the performance of static pages or the full flexibility of server rendering for each page of your website.
+
+In this post, we'll go over how Hybrid Rendering works by exploring just a few of the new features and performance improvements made possible in Astro 2.0.
+
+## How does `astro build` work?
+
+We'll simplify things a bit for clarity, but we first need to understand that Astro's build process happens in multiple stages. First, Astro uses [Vite](https://vitejs.dev) to generate a server-side JavaScript bundle. While this bundle is being generated, we perform static analysis on your source code to generate a **client manifest**. The output of this bundle includes:
```

Comment:
Also there’s a slight disruption to the flow of this paragraph because it goes:

1. “generate a bundle”
2. “while bundle, client manifest”
3. “bundle includes”

Could it make sense to remove the client manifest mention here and just add a mention of “static analysis” to its bullet below?

The bolding of “client manifest” particularly threw me because it makes it seem like the most important thing in that paragraph but at this point I have no idea what it is.

Reply:
Agreed! I removed the bolding and will definitely rephrase this one.